src/main/java/frc/robot/shooter/ShooterSubsystem.java

```diff
@@ -0,0 +1,48 @@
+package frc.robot.shooter;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import au.grapplerobotics.LaserCan;
+
+import frc.robot.Constants;
+
+public class ShooterSubsystem extends SubsystemBase {
+    private CANSparkMax indexerMotor = new CANSparkMax(Constants.Shooter.INDEXER_MOTOR_ID, CANSparkMax.MotorType.kBrushless);
+    private CANSparkMax shooterMotorOne = new CANSparkMax(Constants.Shooter.SHOOTER_MOTOR_ONE_ID, CANSparkMax.MotorType.kBrushless);
+    private CANSparkMax shooterMotorTwo = new CANSparkMax(Constants.Shooter.SHOOTER_MOTOR_TWO_ID, CANSparkMax.MotorType.kBrushless);
+    private RelativeEncoder shooterMotorOneEncoder = shooterMotorOne.getEncoder();
+    private RelativeEncoder shooterMotorTwoEncoder = shooterMotorTwo.getEncoder();
+
+    private LaserCan laserCAN = new LaserCan(Constants.Shooter.LASER_CAN_ID);
+    
+    private double targetShooterSpeed = 0;
+
+    public boolean isIndexerLoaded() {
+        LaserCan.Measurement measurement = laserCAN.getMeasurement(); // CREDIT NOAH WEISHAN
+        return measurement.distance_mm < Constants.Shooter.INDEXER_NOTE_DETECTION_RANGE;
+    }
+
+    public boolean isShooterReady() {
+        double shooterOneSpeedOffset = shooterMotorOneEncoder.getVelocity() - targetShooterSpeed;
+        double shooterTwoSpeedOffset = shooterMotorTwoEncoder.getVelocity() - targetShooterSpeed;
+        return Math.abs(shooterOneSpeedOffset) < Constants.Shooter.TARGET_SPEED_ERROR_MARGIN && Math.abs(shooterTwoSpeedOffset) < Constants.Shooter.TARGET_SPEED_ERROR_MARGIN;
+    }
+
+    public void setTargetShooterSpeed(double targetSpeed) {
+        shooterMotorOne.set(targetSpeed);
+        shooterMotorTwo.set(targetSpeed);
+        targetShooterSpeed = targetSpeed; // TODO: NEEDS TO CONVERT RATIO TO RPM
```

Comment:
You might need to use a PID controller to get this to the desired state. You would probably use the PID controller built in to REVLib. https://codedocs.revrobotics.com/java/com/revrobotics/sparkmaxpidcontroller. You can meet with me and Michael to talk about this more.

Reply:
One thing to remember is last year we learned how to use trapezoidal motion profiles to create much smoother control. This puts much less stress on the mechanical components. We used the wpilib trapezoidal motion profile but SparkMax also supports this. See “smartmotion” related methods. Might not be critical for first iteration of this subsystem, but these were a big improvement we made last year so good to continue to use them.